db/error_handler.cc

```diff
@@ -0,0 +1,106 @@
+};
+
+std::map<std::tuple<BackgroundErrorReason, Status::Code, bool>, Status::Severity> DefaultErrorSeverityMap = {
+  // Errors during BG compaction
+  {std::make_tuple(BackgroundErrorReason::kCompaction, Status::Code::kCorruption, true), Status::Severity::kUnrecoverableError},
+  {std::make_tuple(BackgroundErrorReason::kCompaction, Status::Code::kCorruption, false), Status::Severity::kNoError},
+  {std::make_tuple(BackgroundErrorReason::kCompaction, Status::Code::kIOError, true), Status::Severity::kFatalError},
+  {std::make_tuple(BackgroundErrorReason::kCompaction, Status::Code::kIOError, false), Status::Severity::kNoError},
+  // Errors during BG flush
+  {std::make_tuple(BackgroundErrorReason::kFlush, Status::Code::kCorruption, true), Status::Severity::kUnrecoverableError},
+  {std::make_tuple(BackgroundErrorReason::kFlush, Status::Code::kCorruption, false), Status::Severity::kNoError},
+  {std::make_tuple(BackgroundErrorReason::kFlush, Status::Code::kIOError, true), Status::Severity::kFatalError},
+  {std::make_tuple(BackgroundErrorReason::kFlush, Status::Code::kIOError, false), Status::Severity::kNoError},
+  // Errors during Write
+  {std::make_tuple(BackgroundErrorReason::kWriteCallback, Status::Code::kCorruption, true), Status::Severity::kUnrecoverableError},
+  {std::make_tuple(BackgroundErrorReason::kWriteCallback, Status::Code::kCorruption, false), Status::Severity::kNoError},
+  {std::make_tuple(BackgroundErrorReason::kWriteCallback, Status::Code::kIOError, true), Status::Severity::kFatalError},
+  {std::make_tuple(BackgroundErrorReason::kWriteCallback, Status::Code::kIOError, false), Status::Severity::kNoError},
+};
+
+std::map<std::tuple<BackgroundErrorReason, bool>, Status::Severity> DefaultReasonMap = {
+  // Errors during BG compaction
+  {std::make_tuple(BackgroundErrorReason::kCompaction, true), Status::Severity::kFatalError},
+  {std::make_tuple(BackgroundErrorReason::kCompaction, false), Status::Severity::kNoError},
+  // Errors during BG flush
+  {std::make_tuple(BackgroundErrorReason::kFlush, true), Status::Severity::kFatalError},
+  {std::make_tuple(BackgroundErrorReason::kFlush, false), Status::Severity::kNoError},
+  // Errors during Write
+  {std::make_tuple(BackgroundErrorReason::kWriteCallback, true), Status::Severity::kFatalError},
+  {std::make_tuple(BackgroundErrorReason::kWriteCallback, false), Status::Severity::kFatalError},
+  // Errors during Memtable update
+  {std::make_tuple(BackgroundErrorReason::kMemTable, true), Status::Severity::kFatalError},
+  {std::make_tuple(BackgroundErrorReason::kMemTable, false), Status::Severity::kFatalError},
+};
+
+Status ErrorHandler::SetBGError(const Status& bg_err, BackgroundErrorReason reason) {
+  if (bg_err.ok()) {
+    return Status::OK();
+  }
+
+  auto paranoid = db_options_.paranoid_checks;
+  Status::Severity sev = Status::Severity::kNoError;
```

Comment:
Is default no error risky? How about set a default that stops the DB? In that case, in case the maps are not covering all the possible cases, at least we are stopping the DB.

Reply:
Yes, we can set the default to fatal error to be on the safe side.